compiler/src/scala/quoted/runtime/impl/Matcher.scala

```diff
@@ -329,23 +329,30 @@ object Matcher {
                 s""">>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
                    |Scrutinee
                    |  ${scrutinee.show}
-                   |
-                   |${scrutinee.showExtractors}
-                   |
                    |did not match pattern
                    |  ${pattern.show}
                    |
-                   |${pattern.showExtractors}
-                   |
                    |with environment: ${summon[Env]}
                    |
+                   |Scrutinee: ${scrutinee.showExtractors}
+                   |Pattern: ${pattern.showExtractors}
                    |
                    |""".stripMargin)
             notMatched
         }
       }
     end extension
 
+    /** Does the scrutenne symbol match the pattern symbol? It matches if:
+     *   - They are the same symbol
+     *   - The scrutinee has is in the environment and they are equivalent
+     *   - The scrutinee overrides the symbol of the pattern
+     */
+    private def symbolMatch(scrutinee: Symbol, pattern: Symbol)(using Env): Boolean =
+      scrutinee == pattern
+      || summon[Env].get(scrutinee).contains(pattern)
+      || scrutinee.allOverriddenSymbols.contains(pattern)
```

Comment:
The semantics still seem incomplete in this implementation.

Consider:

```scala
trait A { def foo: Int }
trait B extends A { def foo: Int = 1 }

val e0: Expr[A] = '{ new B }
val e1: Expr[Int] = '{ $e0.foo }
```

The quoted expression `e1` will not match a pattern like `'{ ($x: B).foo }`, but it should.